core/src/jvmTest/kotlin/com/ustadmobile/core/controller/ReportDetailPresenterTest.kt

```diff
@@ -0,0 +1,191 @@
+            toDate = DateTime(2019, 6, 11).unixMillisLong
+            reportFilterList = listOf(
+                    ReportFilter().apply {
+                        entityUid = 100
+                        entityType = ReportFilter.PERSON_FILTER
+                    },
+                    ReportFilter().apply {
+                        entityUid = 200
+                        entityType = ReportFilter.VERB_FILTER
+                    },
+                    ReportFilter().apply {
+                        entityUid = 300
+                        entityType = ReportFilter.CONTENT_FILTER
+                    }
+            )
+        }
+
+        val presenterArgs = mapOf(ARG_ENTITY_JSON to Json.stringify(ReportWithFilters.serializer(), testEntity))
+        val presenter = ReportDetailPresenter(context,
+                presenterArgs, mockView, mockLifecycleOwner,
+                systemImplRule.systemImpl, clientDbRule.db, clientDbRule.repo,
+                clientDbRule.accountLiveData)
+
+        presenter.onCreate(null)
+
+
+        presenter.handleOnClickAddFromDashboard(testEntity)
+
+        val existingEntitiesLive = clientDbRule.db.reportDao.findAllLive()
+        val entitySaved = runBlocking {
+            existingEntitiesLive.waitUntil { it.size == 1 }
+        }.getValue()!!.first()
+        Assert.assertEquals("Entity was saved to database", "New Report Title",
+                entitySaved.reportTitle)
+
+
+
+
+    }
+
+    @Test
+    fun givenReportExists_whenHandleOnClickEditCalled_thenSystemImplGoToEditViewIsCalled() {
+        val testEntity = ReportWithFilters().apply {
+            //set variables here
+            chartType = Report.BAR_CHART
+            xAxis = Report.MONTH
+            fromDate =  DateTime(2019, 4, 10).unixMillisLong
+            toDate = DateTime(2019, 6, 11).unixMillisLong
+            reportFilterList = listOf(
+                    ReportFilter().apply {
+                        entityUid = 100
+                        entityType = ReportFilter.PERSON_FILTER
+                    },
+                    ReportFilter().apply {
+                        entityUid = 200
+                        entityType = ReportFilter.VERB_FILTER
+                    },
+                    ReportFilter().apply {
+                        entityUid = 300
+                        entityType = ReportFilter.CONTENT_FILTER
+                    }
+            )
+            reportUid = clientDbRule.db.reportDao.insert(this)
+        }
+        val presenterArgs = mapOf(ARG_ENTITY_UID to testEntity.reportUid.toString())
+        val presenter = ReportDetailPresenter(context,
+                presenterArgs, mockView, mockLifecycleOwner,
+                systemImplRule.systemImpl, clientDbRule.db, clientDbRule.repo,
+                clientDbRule.accountLiveData)
+
+        presenter.onCreate(null)
+
+        //wait for the entity value to be set
+        mockView.captureLastEntityValue(120000)
```

Comment:
Timeout should be shorter - e.g. 5 seconds is long enough for it to load one entity from the database. Otherwise if the test is failing, this could cause long delays.